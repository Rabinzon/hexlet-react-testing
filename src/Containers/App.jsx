import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'react-jss';
import { pure, compose } from 'recompose';
import Tabs from './Tabs';

const styles = {
  root: {
    color: 'red',
  },
};

const tabTitles = [
  'first tab',
  'second tab',
];

const tabContents = [
  'first tab content',
  'second tab content',
];

const MyComponent = (props) => {
  const { classes } = props;

  return (
    <div className={classes.root}>
      <p>
        Hello world!
      </p>
      <Tabs
        titles={tabTitles}
        contents={tabContents}
      />
    </div>
  );
};

MyComponent.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default compose(
  pure,
  withStyles(styles),
)(MyComponent);
