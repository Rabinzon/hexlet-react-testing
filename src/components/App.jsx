import React from 'react';
import PropTypes from 'prop-types';
import Tabs from './Tabs';

const tabTitles = [
  'first tab',
  'second tab',
];

const tabContents = [
  'first tab content',
  'second tab content',
];

const App = ({ titles = tabTitles, contents = tabContents }) => (
  <div>
    <p>
      Hello world!
    </p>
    <Tabs
      titles={titles}
      contents={contents}
    />
  </div>
);

App.propTypes = {
  titles: PropTypes.array,
  contents: PropTypes.array,
};


export default App;
