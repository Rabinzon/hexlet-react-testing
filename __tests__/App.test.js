import React from 'react';
import { mount } from 'enzyme';
import 'react-log-state';

import App from '../src/components/App';

const tabContents = [
  'first tab content',
  'second tab content',
];

describe('application', () => {
  /* it('should render components tree', () => {
    const wrapper = render(<App contents={tabContents} />);
    expect(wrapper).toMatchSnapshot();
  }); */

  it('should change tab', () => {
    let content;

    const wrapper = mount(<App />);

    const controls = wrapper.find('[data-test="tab-control"]').hostNodes();
    const firstTab = controls.at(0);
    const secondTab = controls.at(1);

    content = wrapper.find('[data-test="tabs-content"]');
  
    expect(content).toHaveText(tabContents[0]);
    secondTab.simulate('click');
    expect(content).toHaveText(tabContents[1]);
    firstTab.simulate('click');
    expect(content).toHaveText(tabContents[0]);
  });

  it('should add new tab', () => {
    const wrapper = mount(<App />);
    let $contents = wrapper.find('[data-test="tabs-content"]').children();

    const $addBtn = wrapper.find('button[data-test="add-tab"]');
    const $removeBtn = wrapper.find('button[data-test="remove-tab"]');

    expect($contents).toHaveLength(tabContents.length);
    $addBtn.simulate('click');
    $contents = wrapper.find('[data-test="tabs-content"]').children();
    expect($contents).toHaveLength(tabContents.length + 1);

    $removeBtn.simulate('click');
    $contents = wrapper.find('[data-test="tabs-content"]').children();
    expect($contents).toHaveLength(tabContents.length);
  });
});
