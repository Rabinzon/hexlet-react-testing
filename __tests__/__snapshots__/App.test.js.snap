// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application should change tab 1`] = `
<div
  data-test="tabs-content"
>
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    data-test="tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  >
    first tab content
  </div>
  <div
    aria-labelledby="react-tabs-6"
    class="react-tabs__tab-panel"
    data-test="tab-panel"
    id="react-tabs-7"
    role="tabpanel"
  />
</div>
`;

exports[`application should change tab 2`] = `
<div>
  <p>
    Hello world!
  </p>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-control"
        id="react-tabs-4"
        role="tab"
      >
        first tab
      </li>
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-control"
        id="react-tabs-6"
        role="tab"
        tabindex="0"
      >
        second tab
      </li>
    </ul>
    <div
      data-test="tabs-content"
    >
      <div
        aria-labelledby="react-tabs-4"
        class="react-tabs__tab-panel"
        data-test="tab-panel"
        id="react-tabs-5"
        role="tabpanel"
      />
      <div
        aria-labelledby="react-tabs-6"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        data-test="tab-panel"
        id="react-tabs-7"
        role="tabpanel"
      >
        second tab content
      </div>
    </div>
  </div>
</div>
`;

exports[`application should change tab 3`] = `
<div
  data-test="tabs-content"
>
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    data-test="tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  >
    first tab content
  </div>
  <div
    aria-labelledby="react-tabs-6"
    class="react-tabs__tab-panel"
    data-test="tab-panel"
    id="react-tabs-7"
    role="tabpanel"
  />
</div>
`;

exports[`application should render components tree 1`] = `
<div>
  <p>
    Hello world!
  </p>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-control"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        first tab
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-control"
        id="react-tabs-2"
        role="tab"
      >
        second tab
      </li>
    </ul>
    <div
      data-test="tabs-content"
    >
      <div
        aria-labelledby="react-tabs-0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        data-test="tab-panel"
        id="react-tabs-1"
        role="tabpanel"
      >
        first tab content
      </div>
      <div
        aria-labelledby="react-tabs-2"
        class="react-tabs__tab-panel"
        data-test="tab-panel"
        id="react-tabs-3"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;
